### ISSUE TITLE
Statsd (tbc. if more extras) tries to install provider package where it is missing

### ISSUE THREAD

**Issue by jedcunningham** [apache/airflow#12255]
**Apache Airflow version**: 2.0.0b2

**What happened**:

Starting in 2.0.0b2, extras that aren't also providers like `statsd` prevents Airflow from being installed.

**How to reproduce it**:
```
$ docker run --rm -ti python:3.6 bash
# pip install apache-airflow[statsd]==2.0.0b2
Collecting apache-airflow[statsd]==2.0.0b2                 
  Downloading apache_airflow-2.0.0b2-py3-none-any.whl (4.5 MB)                    
...
Collecting statsd<4.0,>=3.3.0; extra == "statsd"          
  Downloading statsd-3.3.0-py2.py3-none-any.whl (11 kB)
ERROR: Could not find a version that satisfies the requirement apache-airflow-providers-statsd; extra == "statsd" (from apache-airflow[statsd]==2.0.0b2) (from versions: none)         
ERROR: No matching distribution found for apache-airflow-providers-statsd; extra == "statsd" (from apache-airflow[statsd]==2.0.0b2)
```

I believe this is from https://github.com/apache/airflow/pull/12233

**Comment by potiuk** [comment]
I was quite sure I tested it :(

**Comment by potiuk** [comment]
I wiill double-check and fix it

**Comment by potiuk** [comment]
Just to clarify - Indeed I tested it. This is just a problem with "statsd" which I mistakenly marked as "provider" extra. @jedcunningham  - other extras like "kerberos" and "ldap" should work fine (as they are not mistakenly marked as having providers).  Could you please double check if those work ?

**Comment by jedcunningham** [comment]
Yes, those do work (sorry, I generalized without checking others).

**Comment by potiuk** [comment]
Cool ! :) No problem. I wil add automated check for that before I fix it :).

**Comment by potiuk** [comment]
> and how to fix this?

Apparently it has been fixed in 2.0.0-beta3, so unless you are using 2.0.0-beta2 (highly unlikely, what you are experiencing is a completely different issue. Which I suggest you to use regular channels (slack/github discussions/github issues) and provide description of your problem - sufficient enough (and including all the information that might be helpful for those looking at trying to help you in their free time).

Asking questions like that on issue that has been apparently solved 3 years ago, is kinda pointless.
